LinkedList.h

```diff
@@ -57,7 +58,7 @@ class LinkedList{
 		Adds a T object in the end of the LinkedList;
 		Increment _size;
 	*/
-	virtual bool add(T);
+    virtual bool add(T&);
```

Comment:
I think this will cause incompatibility with major usage.. In the past, it was by reference, but was changed back to the Type in order to support pointers